tests/main.nf.test

```diff
@@ -8,30 +8,41 @@ nextflow_workflow {
 
         when {
             params {
-                input = "samplesheet-staph.csv"
-                platform = "illumina"
+                input = "https://github.com/phac-nml/mikrokondo/raw/integration-testing/tests/data/samplesheets/samplesheet-campy-staph.csv"
                 outdir = "results"
-                mash.mash_sketch = "campy-staph-ecoli.msh"
-                r_contaminants.mega_mm2_idx = "campy.mmi"
+
+                platform = "illumina"
+
+                mash { mash_sketch = "https://github.com/phac-nml/mikrokondo/raw/integration-testing/tests/data/databases/campy-staph-ecoli.msh" }
+                r_contaminants { mega_mm2_idx = "https://github.com/phac-nml/mikrokondo/raw/integration-testing/tests/data/databases/campy.mmi" }
+
                 skip_bakta = true
                 skip_staramr = true
                 skip_mobrecon = true
                 skip_checkm = true
                 skip_raw_read_metrics = true
                 skip_polishing = true
-                max_memory = "4.GB"
-                max_cpus = 2
             }
             workflow {
                 """
-                // define inputs of the workflow here. Example:
-                // input[0] = file("test-file.txt")
+                // define inputs of the workflow here.
                 """
             }
         }
 
         then {
             assert workflow.success
+            assert path("$launchDir/results").exists()
+
+            // parse output json file
+            def json_string = path("$launchDir/results/SummaryReport/final_report.json").readLines().join("\n")
```

Comment:
I think instead of reading lines directly and parsing as JSON, you could switch to:

```
path("$launchDir/results/SummaryReport/final_report.json").json
```

I believe this will return the file parsed as a JSON object. I'm not sure the exact return type, but I suspect it will create a JsonSlurper object and parse the JSON internally and will return the results of that. Could you test this out?

Reply:
Good insight. I didn't know it could be done easier like this. Updated in e62432396637612277b685e494e39139a8a3748b